src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java

```diff
@@ -85,13 +85,18 @@ public static DetailAST parse(FileContents contents)
         filter.hide(TokenTypes.SINGLE_LINE_COMMENT);
         filter.hide(TokenTypes.BLOCK_COMMENT_BEGIN);
 
-        final GeneratedJavaRecognizer parser = new GeneratedJavaRecognizer(filter);
+        final GeneratedJavaRecognizer parser = new GeneratedJavaRecognizer(filter) {
+            @Override
+            public void reportError(RecognitionException ex) {
+                throw new IllegalStateException(ex);
```

Comment:
`reportError` has no throws clause, so we can only throw runtime exceptions here.